drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -755,7 +755,7 @@ static void handle_fmt_changed(struct bcm2835_codec_ctx *ctx,
 	q_data = get_q_data(ctx, V4L2_BUF_TYPE_VIDEO_CAPTURE);
 	q_data->crop_width = format->es.video.crop.width;
 	q_data->crop_height = format->es.video.crop.height;
-	q_data->bytesperline = format->es.video.crop.width;
+	q_data->bytesperline = format->es.video.width;
```

Comment:
I'm confused by this one, but it may just be the wording.
> bytesperline was wrong if width is aligned

If width is aligned, then ```crop.width``` and ```width``` are the same value.

Reviewing the context, this is totally wrong should the format be anything other than YUV because it doesn't take into account that there could be more than 1 byte per pixel.
What is needed is a call to ```get_bytesperline();```. I don't believe the firmware will ever change the colour format, so q_data->fmt is still valid as the format. If passing ```format->es.video.width``` as the width then it should always be aligned, but get_bytesperline will enforce that anyway.

Thanks for picking this up though - I'm now wondering if this is why ffmpeg was having issues and I had to patch it.

Reply:
get_bytesperline() is also good. Now are in bytesperline a wrong value. For example a video width 720 have a bytesperline 736. But in bytesperline are 720. Do you make a patch or I shall do it?